src/sentry/integrations/slack/utils/users.py

```diff
@@ -18,31 +22,58 @@
 
 
 def get_users(integration: Integration, organization: Organization) -> Sequence[Mapping[str, Any]]:
-    client = SlackClient(integration_id=integration.id)
-
     user_list = []
     next_cursor = None
-    for i in range(SLACK_GET_USERS_PAGE_LIMIT):
+
+    if features.has("organizations:slack-sdk-get-users", organization):
+        sdk_client = SlackSdkClient(integration_id=integration.id)
+
         try:
-            next_users = client.get(
-                "/users.list",
-                params={"limit": SLACK_GET_USERS_PAGE_SIZE, "cursor": next_cursor},
-            )
-        except ApiError as e:
+            users = sdk_client.users_list(cursor=next_cursor, limit=SLACK_GET_USERS_PAGE_SIZE)
+        except SlackApiError as e:
             logger.info(
-                "post_install.fail.slack_users.list",
+                "slack.post_install.get_users.error",
                 extra={
                     "error": str(e),
                     "organization": organization.slug,
                     "integration_id": integration.id,
                 },
             )
-            break
-        user_list += next_users["members"]
+            return []
```

Comment:
this function is called inside an async task to link slack user identities inside integration post_install. we could error loudly and have retries on the task?

Reply:
Hmmm, would it be better to return `None` and update the function contract? Might also be good to just reraise and let the caller decide what it wants to do instead of implicitly forcing it